osclib/stagingapi.py

```diff
@@ -998,16 +998,17 @@ def map_ring_package_to_subject(self, project, pkg):
         """
         # it's actually a pretty stupid algorithm, but it might become more complex later
 
-        if project.endswith(':DVD'):
-            return project  # not yet
+        # assuming it is in adi staging, workaround for https://progress.opensuse.org/issues/9646
+        if self.is_adi_project(project) and self.item_exists(project, pkg):
+            return project
 
         ring_dvd = '{}:2-TestDVD'.format(self.crings)
-        if self.ring_packages.get(pkg) == ring_dvd:
-            if not self.item_exists(project + ":DVD") and self.item_exists(project, pkg):
-                # assuming it is in adi staging, workaround for https://progress.opensuse.org/issues/9646
-                return project
-            else:
-                return project + ":DVD"
+        if self.ring_packages.get(pkg) == ring_dvd and not project.endswith(':DVD'):
```

Comment:
Something I've done elsewhere that I find cleaner than building a string to compare against to determine ring project is the following.

```python
if self.rings.index(self.ring_packages.get(pkg)) == 2:
```

No extra string creation with a pattern that is now hard-coded in another place and the rings line up with array indexes.

An example in [`StagingAPI.ring_archs()` line 1552](https://github.com/nilxam/osc-plugin-factory/blob/c2808dd22cfbe107633954d341dee13d09627cbb/osclib/stagingapi.py#L1552).